### PR TITLE
[mqtt] Add a new checkin pipeline for mqtt broker

### PR DESCRIPTION
**Pull request by dmolokanov**

### CODE REVIEW

File: builds/checkin/dotnet.yaml [hunk 1]
```diff
@@ -9,14 +9,14 @@ jobs:
...
+    displayName: Check pipeline preconditions (changes ARE NOT in either edgelet, docs, or mqtt folder)
     pool:
       vmImage: "ubuntu-16.04"
     steps:
       - bash: |
-          git log -m -1 --name-only --first-parent --pretty="" | egrep -i -v '^(edgelet|doc)'
+          git log -m -1 --name-only --first-parent --pretty="" | egrep -i -v '^(edgelet|doc|mqtt)'
           if [[ $? == 0 ]]; then
-            echo "Detected changes outside of edgelet and docs folders"
+            echo "Detected changes outside of edgelet, docs or mqtt folders"
```
Comment (philipktlin) [inline builds/checkin/dotnet.yaml]:
```suggestion
            echo "Detected changes outside of edgelet, docs and mqtt folders"
```

File: builds/checkin/mqtt.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  ################################################################################
+  - job: check_run_pipeline
+    ################################################################################
+    displayName: Check pipeline preconditions (changes ARE in builds or mqtt)
+    pool:
+      vmImage: "ubuntu-16.04"
+    steps:
+      - bash: |
+          git log -m -1 --name-only --first-parent --pretty="" | egrep -i '^(builds|mqtt)'
```
Comment (philipktlin) [inline builds/checkin/mqtt.yaml]:
```suggestion
          git log -m -1 --name-only --first-parent --pretty="" | egrep -i '(builds|mqtt)'
```

File: builds/checkin/mqtt.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          if [[ $? == 0 ]]; then
+            echo "Detected changes inside builds or mqtt folders"
+            echo "##vso[task.setvariable variable=RUN_PIPELINE;isOutput=true]TRUE"
+          fi
+        displayName: Check changes in sources
+        name: check_files
+
+  ################################################################################
+  - job: linux_amd64
```
Comment (philipktlin) [inline builds/checkin/mqtt.yaml]: will cover ARM32 and ARM64 in next PR?

Comment (dmolokanov) [reply builds/checkin/mqtt.yaml]: I tried to simplify this pipeline. Eventually we add support for arm based ones as well. Not now

**Pull request closed without merging**